api/src/test/java/com/messagebird/RequestSignerTest.java

```diff
@@ -16,51 +17,95 @@
         String expectedSignature = "LISw4Je7n0/MkYDgVSzTJm8dW6BkytKTXMZZk1IElMs=";
         Request request = new Request("1544544948", "", getBytes(""));
 
         assertTrue(requestSigner.isMatch(expectedSignature, request));
     }
 
-    private void testIsMatchWithData() {
+    @Test
+    public void testIsMatchWithData() {
         RequestSigner requestSigner = new RequestSigner(getBytes("secret"));
         String expectedSignature = "p2e20OtAg39DEmz1ORHpjQ556U4o1ZaH4NWbM9Q8Qjk=";
         Request request = new Request("1544544948", "", getBytes("{\"a key\":\"some value\"}"));
 
         assertTrue(requestSigner.isMatch(expectedSignature, request));
     }
 
-    private void testIsMatchWithQueryParams() {
+    @Test
+    public void testIsMatchWithQueryParams() {
         RequestSigner requestSigner = new RequestSigner(getBytes("secret"));
         String expectedSignature = "Tfn+nRUBsn6lQgf6IpxBMS1j9lm7XsGjt5xh47M3jCk=";
         Request request = new Request("1544544948", "abc=foo&def=bar", getBytes(""));
 
         assertTrue(requestSigner.isMatch(expectedSignature, request));
     }
 
-    private void testIsMatchWithDataAndQueryParams() {
+    @Test
+    public void testIsMatchWithShuffledQueryParams() {
+        RequestSigner requestSigner = new RequestSigner(getBytes("secret"));
+        String expectedSignature = "Tfn+nRUBsn6lQgf6IpxBMS1j9lm7XsGjt5xh47M3jCk=";
+        Request request = new Request("1544544948", "def=bar&abc=foo", getBytes(""));
+
+        assertTrue(requestSigner.isMatch(expectedSignature, request));
+    }
+
+    @Test
+    public void testIsMatchWithDataAndQueryParams() {
         RequestSigner requestSigner = new RequestSigner(getBytes("other-secret"));
         String expectedSignature = "orb0adPhRCYND1WCAvPBr+qjm4STGtyvNDIDNBZ4Ir4=";
         Request request = new Request("1544544948", "abc=foo&def=bar", getBytes("{\"a key\":\"some value\"}"));
 
         assertTrue(requestSigner.isMatch(expectedSignature, request));
     }
 
-    private void testIsNotMatch() {
+    @Test
+    public void testIsNotMatch() {
         RequestSigner requestSigner = new RequestSigner(getBytes("secret"));
         String expectedSignature = "";
         Request request = new Request("1544544948", "abc=foo&def=bar", getBytes("{\"a key\":\"some value\"}"));
 
         assertFalse(requestSigner.isMatch(expectedSignature, request));
     }
+
+    @Test
+    public void testWithRealSignature() {
+        /*
+         * Here we use real signature from MessageBird webhook call
+         */
+
+        RequestSigner requestSigner = new RequestSigner(getBytes("Wb3N9gKeFf8ZoCzlOb5lJSic7bHLUcSu"));
+        String requestSignature = "vN48KugM8Db3rNYNfhBRLJWqqTm2LQotRMHAlPCE3fc=";
+        String requestTimestamp = "1547036603";
+        String requestParams = "id=57db52e04e2f4001b555f79813a0f503&mccmnc=20409&ported=0&recipient=31687740260&reference=curl&status=delivered&statusDatetime=2019-01-09T12%3A23%3A23%2B00%3A00";
```

Comment:
Are you ok with leaking your phone number here?

Reply:
Oh! Thank you very much! God bless, it's a temporary lycamobile number :D